### ISSUE TITLE
Finder Extension on macOS 15 Sequoia

### ISSUE THREAD

**Issue by aonez** [aonez/Keka#1464]
Seems Finder Extensions are no more on upcoming macOS 15 Sequoia, at least on the first beta.

Posted on https://forums.developer.apple.com/forums/thread/756683 to see if anyone else can confirm.

---

# ✅✅✅ Fixed in the upcoming macOS 15.2

https://developer.apple.com/forums/thread/756711?answerId=812625022#812625022

---

## Workaround for macOS 15.0 - 15.1

There is a fix implemented in the Keka Helper v1.2.0 since Keka v1.4.4:

👉 https://www.keka.io/finderextensionsequoia

**Comment by Rishi-Sudhakar** [comment]
Can confirm, its no longer there. 
<img width="827" alt="Screenshot 2024-06-15 at 5 39 39 PM" src="https://github.com/aonez/Keka/assets/79398572/ebd52fb0-0bbc-4b1a-8e67-2c2d6aca25c2">

**Comment by Kief5555** [comment]
Yep, I cannot enable nor see Keka in extensions.

**Comment by JoshuaFern** [comment]
There's a simple workaround for now. Simply run `pluginkit -mAD -p com.apple.FinderSync -vvv` which produces an output similar to:
```
     com.aone.keka.KekaFinderIntegration(1.4.3)
	            Path = /Applications/Keka.app/Contents/PlugIns/KekaFinderIntegration.appex
	            UUID = AB3031DB-D94B-44D0-AD58-EA935ACEA96C
	       Timestamp = 2024-08-10 21:36:21 +0000
	             SDK = com.apple.FinderSync
	   Parent Bundle = /Applications/Keka.app
	    Display Name = Keka Finder Integration
	      Short Name = Keka Finder Integration
	     Parent Name = Keka
	        Platform = macOS
```
Then you can run `pluginkit -e "use" -u "AB3031DB-D94B-44D0-AD58-EA935ACEA96C"` to enable the extension.

**Comment by aonez** [comment]
Thanks @JoshuaFern, this workaround will be implemented using the Keka Helper in the next revision.

**Comment by gingerbeardman** [comment]
There's an app for that. 

https://zigz.ag/FinderSyncer/

**Comment by aonez** [comment]
Sorry I did not informed here that this is implemented in the Keka Helper v1.2.0 since Keka v1.4.4:
https://www.keka.io/finderextensionsequoia

**Comment by milke** [comment]
@gingerbeardman, I'll be free to add that [FinderSyncer](https://zigz.ag/FinderSyncer) should not really be the solution, but a temporary workaround instead. I'm afraid that everything happening with Finder Sync settings in recent versions of System Settings application is a sign of slow deprecation and discontinuation of Finder Sync extensions, especially as Apple pushes **File Provider API** for "_applications which synchronize the contents of a local folder with a remote data source_". And they claim that was the initial use case for Finder Sync extensions.

I'd urge @aonez and all other developers distributing Finder Sync extensions to read discussion on [this thread](https://developer.apple.com/forums/thread/756711) on Apple Devs forums (after all, that thread was spawn off of another one created by @aonez himself, [this one](https://developer.apple.com/forums/thread/756683)), chime in and (try to) let Apple folks know that if they deprecate Finder Sync extensions, whey will break a lot of third party applications. I have no illusion Apple would be very stressed about it, they keep doing whatever they want for the past 8 - 10 years anyway, but at least we can try.

**Comment by gingerbeardman** [comment]
Of course.

**Comment by aonez** [comment]
This is fixed in the upcoming macOS 15.2:
https://developer.apple.com/forums/thread/756711?answerId=812625022#812625022

Next version of Keka will use the proper way to activate the extension in Sequoia if macOS 15.2 or later or macOS 14.* or older.

**Comment by aonez** [comment]
v1.4.7 can be activated again in the Sequoia Settings. The workaround is only enabled for 15.0-15.1.x.
